src/main/java/com/cloudogu/smeagol/wiki/infrastructure/HistoryResourceAssembler.java

```diff
@@ -0,0 +1,66 @@
+package com.cloudogu.smeagol.wiki.infrastructure;
+
+
+import com.cloudogu.smeagol.wiki.domain.Commit;
+import com.cloudogu.smeagol.wiki.domain.History;
+import com.cloudogu.smeagol.wiki.domain.WikiId;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+
+public class HistoryResourceAssembler extends ResourceAssemblerSupport<History, HistoryResource> {
+
+    public HistoryResourceAssembler() {
+        super(HistoryController.class, HistoryResource.class);
+    }
+
+
+    private final CommitResourceAssembler assembler = new CommitResourceAssembler();
+
+    @Override
+    public HistoryResource toResource(History history) {
+        List<CommitResource> commitsAsResources = new ArrayList<CommitResource>(history.getCommits().size());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS2293)